receiver/http.go

```diff
@@ -56,6 +57,7 @@ type HTTP struct {
 	Certfile      string                        `doc:"Path to certificate file for TLS. If left blank, un-encrypted HTTP is used."`
 	Keyfile       string                        `doc:"Path to key file for TLS."`
 	AcceptableCAs []string                      `doc:"Paths to files containing CAs which are accepted for Client Certificate authentication."`
+	SANDNSName    string                        `doc:"DNS name which has to be present in SAN extension of x509 certificate when using Client Certificate authentication"`
```

Comment:
We could try to work this more into the style of HTTPAuth (line 42-46), as adding struct fields for each of the potential ways to verify (including DNSNames, EmailAddresses, IPAddressess, URIs) could become a bit messy if all added onto the base HTTP struct.

Just a thought.